core/chains/evm/txmgr/orm.go

```diff
@@ -121,34 +120,38 @@ func (o *orm) Close() {
 	o.ctxCancel()
 }
 
-func (o *orm) preloadTxAttempts(txs []EthTx) error {
-	// Preload TxAttempts
-	var ids []int64
-	for _, tx := range txs {
-		ids = append(ids, tx.ID)
-	}
-	if len(ids) == 0 {
-		return nil
-	}
-	var attempts []EthTxAttempt
-	sql := `SELECT * FROM eth_tx_attempts WHERE eth_tx_id IN (?) ORDER BY id desc;`
-	query, args, err := sqlx.In(sql, ids)
-	if err != nil {
-		return err
-	}
-	query = o.q.Rebind(query)
-	if err = o.q.Select(&attempts, query, args...); err != nil {
-		return err
-	}
-	// fill in attempts
-	for _, attempt := range attempts {
-		for i, tx := range txs {
-			if tx.ID == attempt.EthTxID {
-				txs[i].EthTxAttempts = append(txs[i].EthTxAttempts, attempt)
+func (o *orm) preloadTxAttempts(txs []EthTx, qopts ...pg.QOpt) error {
```

Comment:
Why are we adding qopts to this method?

Reply:
this is an internal method so I thought it would be ok. But the reason why is because we want to have preloading tx attempts to be in the same transaction as other methods so typical use case would be passing in the tx pg.Queryer

Reply:
removed 👍